src/pages/o/[orgId]/events/[eventId].tsx

```diff
@@ -0,0 +1,53 @@
+import { 
+    GetServerSideProps,
+    NextPageContext,
+} from 'next';
+
+export const getServerSideProps : GetServerSideProps = async (context : NextPageContext) => {
+    const { orgId } = context.params;
+    const { eventId } = context.params;
+    let allEventsData = [];
+
+    const cRes = await fetch(`http://api.zetk.in/v1/orgs/${orgId}/campaigns`);
+    const cData = await cRes.json();
+
+    for (const obj of cData.data) {
+        const eventsRes = await fetch(`https://api.zetk.in/v1/orgs/${orgId}/campaigns/${obj.id}/actions`);
+        const campaignEvents = await eventsRes.json();
+        allEventsData = allEventsData.concat(campaignEvents.data);
+    }
+
+    const eventData = allEventsData.find(event => event.id == eventId);
```

Comment:
This could be optimized by searching each array as it comes back from the API, instead of concatenating all of them into a large single array and then searching.

If there are three campaigns (A, B, C) returned by the API, each contain 100 actions, and the requested action is the 50th action of campaign B, this code basically does this:

1. Fetch actions from campaign A, store 100 actions
2. Fetch actions from campaign B, concatenate results with A (now 200 actions)
3. Fetch actions from campaign C, concatenate results with AB (now 300 actions)
4. Iterate over ABC array, 150 actions before finding the correct action

If instead the search happened as part of the loop, and the loop stopped if the correct action is found, the operations would instead be:

1. Fetch actions from campaign A
2. Search actions from campaign A (100 inspected, requested action not found)
3. Fetch actions from campaign B
4. Search actions from campaign B (50 inspected, 50th action matches)
5. Break from loop

This saves the request to campaign C, which is not necessary.